exec/java-exec/src/main/resources/rest/index.ftl

```diff
@@ -103,14 +105,14 @@
                 </td>
                 <td id="status" >${drillbit.getState()}</td>
                 <td class="uptime" >Not Available</td>
-                  <td>
-                <#if ( model.shouldShowAdminInfo() &&  ( drillbit.isCurrent() || ( !model.isAuthEnabled() && location.protocol != "https" ))) >
-                      <button type="button" id="shutdown" onClick="shutdown($(this), '${drillbit.getAddress()}:${drillbit.getHttpPort()}');">
-                <#else>
+                <td>
+                  <#if ( model.shouldShowAdminInfo() || !model.isAuthEnabled() || drillbit.isCurrent() ) >
```

Comment:
This check doesn't look correct to me. `shouldShowAdminInfo` checks If the logged in user is admin or not. So for admin user it will show the shutdown button in if block even for remote Drillbit which is not the intention.

Reply:
Yes, and (unfortunately), I can't check for 'https' protocol within freemarker. The check for this is taken care of at [line 317](https://github.com/kkhatua/drill/blob/ab3e8619c6259803eb362be290a3a3605839a194/exec/java-exec/src/main/resources/rest/index.ftl#L317) of `updateStatusAndShutdown` function, where I check for *HTTPS* and as a current Drillbit.
```javascript
                if (status_map[key] == "ONLINE") {
                    $("#row-"+i).find("#status").text(status_map[key]);
                    <#if ( model.shouldShowAdminInfo() || !model.isAuthEnabled() ) >
                    if ( location.protocol != "https" || ($("#row-"+i).find("#current").html() == "Current") ) {
                      $("#row-"+i).find("#shutdown").prop('disabled',false).css('opacity',1.0).css('cursor','pointer');
                    }
                    </#if>
```